agent/acs/handler/task_manifest_handler.go

```diff
@@ -0,0 +1,280 @@
+	messageBufferTaskStopVerificationAck     chan *ecsacs.TaskStopVerificationAck
+	ctx                                      context.Context
+	taskEngine                               engine.TaskEngine
+	cancel                                   context.CancelFunc
+	saver                                    statemanager.Saver
+	cluster                                  string
+	containerInstanceArn                     string
+	acsClient                                wsclient.ClientServer
+	seqNumTaskManifest                       *int64
+	messageId                                string
+}
+
+// newTaskManifestHandler returns an instance of the taskManifestHandler struct
+func newTaskManifestHandler(ctx context.Context,
+	cluster string, containerInstanceArn string, acsClient wsclient.ClientServer,
+	saver statemanager.Saver, taskEngine engine.TaskEngine, seqNumTaskManifest *int64) taskManifestHandler {
+
+	// Create a cancelable context from the parent context
+	derivedContext, cancel := context.WithCancel(ctx)
+
+	return taskManifestHandler{
+		messageBufferTaskManifest:                make(chan *ecsacs.TaskManifestMessage),
+		messageBufferTaskManifestAck:             make(chan string),
+		messageBufferTaskStopVerificationMessage: make(chan *ecsacs.TaskStopVerificationMessage),
+		messageBufferTaskStopVerificationAck:     make(chan *ecsacs.TaskStopVerificationAck),
+		ctx:                                      derivedContext,
+		cancel:                                   cancel,
+		cluster:                                  cluster,
+		containerInstanceArn:                     containerInstanceArn,
+		acsClient:                                acsClient,
+		taskEngine:                               taskEngine,
+		saver:                                    saver,
+		seqNumTaskManifest:                       seqNumTaskManifest,
+	}
+}
+
+func (taskManifestHandler *taskManifestHandler) handlerFuncTaskManifestMessage() func(
+	message *ecsacs.TaskManifestMessage) {
+	return func(message *ecsacs.TaskManifestMessage) {
+		taskManifestHandler.messageBufferTaskManifest <- message
+	}
+}
+
+func (taskManifestHandler *taskManifestHandler) handlerFuncTaskVerificationMessage() func(
+	message *ecsacs.TaskStopVerificationAck) {
+	return func(message *ecsacs.TaskStopVerificationAck) {
+		taskManifestHandler.messageBufferTaskStopVerificationAck <- message
+	}
+}
+
+func (taskManifestHandler *taskManifestHandler) start() {
+	// Task manifest and it's ack
+	go taskManifestHandler.handleTaskManifestMessage()
+	go taskManifestHandler.sendTaskManifestMessageAck()
+
+	// Task stop verification message and it's ack
+	go taskManifestHandler.sendTaskStopVerificationMessage()
+	go taskManifestHandler.handleTaskStopVerificationAck()
+
+}
+
+func (taskManifestHandler *taskManifestHandler) sendTaskManifestMessageAck() {
+	for {
+		select {
+		case mid := <-taskManifestHandler.messageBufferTaskManifestAck:
```

Comment:
what's mid?

Reply:
It's just a placeholder variable, will change it to a more meaningful name.